### DISCUSSION TITLE
Codebase quality improvements

### DISCUSSION THREAD

**Discussion by AntPolkanov** [SSWConsulting/SSW.Rewards.Mobile#413] [Ideas]
Hey team,

I see a few opportunities to improve the quality of our codebase.
It would be cool if we could:

1. Start using XamlStyler (https://github.com/Xavalon/XamlStyler). The plugin is available for both VisualStudio and Rider. It is super easy to get going; I normally use the default formatting configuration only with a few slight modifications. 
**The benefit** is that our xaml code will be consistently formatted across the project.
2. Automate PropertyChangedEvent calls. I used `Fody.PropertyChanged` before https://github.com/Fody/PropertyChanged which is pretty good. I see that we are already using `CommunityToolkit.Mvvm` which also has this capability to inject PropertyChangedEvent; with it we might go even further and start using `IAsyncRelayCommands`. 
**The benefit** is that we will get rid of the boilerplate associated with the PropertyChangedEvent.

If you guys agree with it, I am happy to implement both tasks.
(cc: @matt-goldman, @JackDevAU)

**Comment by matt-goldman** [comment]
Hey @AntPolkanov these are great suggestions!

> Start using XamlStyler

I have used XamlStyler in the past and ended up throwing it out in frustration. I didn't like the way it formatted my XAML, and while it's supposed to be configurable, I could never get it to work. Even when I could get the formatting the way I wanted it, found it reverted back to it's default. I'd be super happy to use it if you know how to overcome this problem!

> Automate PropertyChangedEvent calls

I've used `Fody.PropertyChanged` before, it's pretty awesome. My preference is to use the CommunityToolkit though; as you note we're already using it and we should maximise this (in other ways you've noted too). Fody is great, but I prefer not to add `PropertyChanged` to _every_ public property in the codebase.

1. Do you want to add issues for these? Once they're on the backlog we can pull them in during sprint planning.

**Comment by matt-goldman-ssw** [comment]
Hey @AntPolkanov 👋

Closing this discussion as we have implemented both suggestions.
